tinycolor.js

```diff
@@ -710,50 +728,143 @@ tinycolor.readability = function(color1, color2) {
 
     return {
         brightness: Math.abs(brightnessA - brightnessB),
-        color: colorDiff
+        color: colorDiff,
+        contrast:(Math.max(luminance1,luminance2)+0.05) / (Math.min(luminance1,luminance2)+0.05)
     };
 };
 
-// `readable`
-// http://www.w3.org/TR/AERT#color-contrast
-// Ensure that foreground and background color combinations provide sufficient contrast.
+// `isReadable`
+// Ensure that foreground and background color combinations meet WCAG guidelines.
+// The third argument is optional. If present, it provides an object containing additional information for WCAG 2.0:
+//      the 'level' property states 'AA' or 'AAA' - if missing or invalid, it defaults to 'AA';
+//      the 'size' property states 'large' or 'small' - if missing or invalid, it defaults to 'small'.
+//
 // *Example*
 //    tinycolor.isReadable("#000", "#111") => false
-tinycolor.isReadable = function(color1, color2) {
+//    tinycolor.isReadable("#000", "#111",{level:"AA",size:"large"}) => false
+
+tinycolor.isReadable = function(color1, color2, wcag2) {
     var readability = tinycolor.readability(color1, color2);
-    return readability.brightness > 125 && readability.color > 500;
+    var wcag2Parms, wcag2ParmsConcat,  out;
+    if (wcag2) {
+        out = false;
+
+        // Herewith the algortihm in easy-to-understand format.
+
+        wcag2Parms = validateWCAG2Parms(wcag2);
+        switch (wcag2Parms.level + wcag2Parms.size) {
+            case "AAsmall":
+                out = readability.contrast >= 4.5;
+                break;
+            case "AAlarge":
+                out = readability.contrast >= 3;
+                break;
+            case "AAAsmall":
+                out = readability.contrast >= 7;
+                break;
+            case "AAAlarge":
+                out = readability.contrast >= 4.5;
+                break;
+        }
+
+        // The optimisation below should run faster (and a quick few iterations of the test suite suggest it does),
```

Comment:
Let's go with the more readable / understandable algorithm.  Could optimize in the future, but I doubt it will be needed